src/java/com/sarality/app/view/list/SearchListComponent.java

```diff
@@ -0,0 +1,49 @@
+package com.sarality.app.view.list;
+
+import android.support.v4.app.FragmentActivity;
+import android.widget.ListView;
+
+import com.sarality.app.view.action.ComponentActionManager;
+
+import java.util.List;
+
+/**
+ * Component Object that searches through the list and updates listview
+ *
+ * @param <T> List Data
+ * @author sunayna@dothat.in sunayna
+ */
+public class SearchListComponent<T> extends ListComponent<T> {
+
+  private final FragmentActivity activity;
+  private final ListRowRenderer<T> rowRenderer;
+  private final List<ListItemFilter<T>> filterList;
```

Comment:
Keep this as a ListItemFilter and if we ever need a Filter of Filters we'll just create a CompositeListItemFilter that takes in a List of ListItemFilters and still implements the interface.